python_modules/libraries/dagster-dg/dagster_dg/cli/init.py

```diff
@@ -149,4 +179,40 @@ def init_command(
         python_environment=DgProjectPythonEnvironment.from_flag(python_environment),
     )
 
-    click.echo("You can create additional projects later by running 'dg scaffold project'.")
+    if workspace:
+        click.echo("You can create additional projects later by running 'dg scaffold project'.")
+
+    venv_path = project_path / ".venv"
+
+    # This already will have occurred during the scaffolding step
+    if python_environment == "uv_managed":
+        run_uv_sync = False
+    # If a venv already exists, only run `--uv-sync` if it is explicitly set.
+    elif venv_path.exists():
+        run_uv_sync = uv_sync is True
+    else:
+        if uv_sync is True:
+            run_uv_sync = True
+        elif uv_sync is None and is_uv_installed():
+            response = click.prompt(
+                "Run `uv sync` to bootstrap a uv.lock and virtual environment (y/n)?", default="y"
+            )
+            run_uv_sync = response.lower() == "y"
+        else:
+            run_uv_sync = False
+
+    if run_uv_sync:
+        click.echo("Running `uv sync`...")
+        with pushd(project_path):
+            subprocess.run(["uv", "sync"], check=True)
+
+        click.echo("\nuv.lock and virtual environment created.")
+        relative_venv_path = venv_path.relative_to(Path.cwd())
+        click.echo(
+            format_multiline_str("""
+            Your project is configured with `project.python_environment.active = true`, so
+            you will need to activate the created virtual environment to use it with your `dg`
+            project. To activate the virtual environment, run:
+        """)
```

Comment:
This is case where I think the ``project.python_environment.active = true`might be too much deal.

"Run `source .venv/bin/activate` to active project's environment"